### DISCUSSION TITLE
HEIC depth information 💩

### DISCUSSION THREAD

**Discussion by Milk-Dream-sRGB** [ImageMagick/ImageMagick#3040] [Help] [answered]
Hi everyone!

I lost few days (overall) to understand how I can do this. And I got it. I can't. I'm sorry for this question, really read all about this problem, and can't find solution for me.
I have .HEIC file from iPhone, captured in portrait mode. And I need extract depth mask from photo for edit in Ps. I use OS Windows 10 and this function is not default, unfortunately, in Ps. I try use Imagemagick to convert in .jpg and then extract depth mask with exiftool command "-b -MPImage" - not work. But if I extract mask from jpg generated from iPhone- all work.

I really ask for help. Thank you!
Unfortunately, I am very weak in command line tools :(

If open this photo with Lightroom (on Windows), Lightroom can show mask, but can't export It.

I found info about "Auxiliary Image Storage", HEIC store additional info in auxiliary image storage, like depth map or alpha channel.. But I can't find this data on file :(

New addition: if create in Lightroom "smart preview" from .heic and open smart preview (smart preview automatically saved in .dng) in camera raw- depth mask available in camera raw! I think, this mean what data about depth stores somewhere in .heic...

I try find answer from 2018 :))
And I'm sorry for language mistakes :(

UPD
But if use command in Lightroom "Export" and export to jpg- information about depth not saves (maybe)

maybe last thing- maybe depth data in HEIC store in text data, not image like in jpg... And if its true, how we can export it in image...

I really try find solution very long time...
Thanks all for helping! 

[sample HEIC file](https://www.dropbox.com/s/kb2ju9s70g57azr/IMG_2656.HEIC?dl=0)

**Comment by snibgo** [comment]
I suggest you supply a link to a sample HEIC file.

**Reply by Milk-Dream-sRGB** [reply]
Yes, you right! Thanks :)

**Comment by snibgo** [comment]
IM and exiftool can't see a depth map in IMG_2656.HEIC. Gimp v2.10.0 can't even open the file.

An internet search suggests that the depth map can be read with heif_convert. Source for that program is distributed with libheif. libheif source is included with ImageMagick Windows source. Some hints are in https://github.com/ImageMagick/libheif. I can build this with autoconf, automake, ./configure, make, make install.

The heif programs work:
```
$ heif-info IMG_2656.HEIC
image: 3024x4032 (id=49), primary
  thumbnail: 240x320
  color profile: prof
  alpha channel: no
  depth channel: yes
 (576x768)
    z-near: undefined
    z-far:  undefined
    d-min:  0.427246
    d-max:  2.443359
    representation: uniform disparity
    disparity_reference_view: 0
```

But:
```
$ heif-convert IMG_2656.HEIC asd.png
File contains 1 images
Could not decode HEIF image: 0: Unsupported feature: Unsupported codec
```
So near yet so far. I am missing something, possibly libde265.

**Reply by Milk-Dream-sRGB** [reply]
"So near yet so far" Yes, I try all things what I can do with some converter and Photoshop on windows. And I feel what answered is near, but nothing goes right :)) This little problem goes from 2018, but I have no idea why this feature not interested to more people :)

"IM and exiftool can't see a depth map in IMG_2656.HEIC. Gimp v2.10.0 can't even open the file."
First thing on windows- to open heic files with some software necessary install _both_ codec HEIC (windows decided to sell it..) and HEIF (free), maybe GIMP not open from this thing.
On begin I doubted too, but I opened this image on folder with only this image in Lightroom and it can show me the mask! :)

"heif_convert" I read about that command, but I can't compiling, unfortunately :(
But this line "depth channel: yes" make me inspired :D If it only way to solve problem I try learn about compile more...

Thank you!!!

**Reply by shahdv** [reply]
I was facing a similar problem, and reached this discussion via a Google search. Another search led me to this [StackExchange page](https://photo.stackexchange.com/questions/105378/how-to-extract-depth-map-from-heif-image-created-with-iphone-x-using-imagemagic), where they use `libheif` to extract that information. 

I installed `libheif` within my Ubuntu subsystem running via WSL on Windows 10 - 
`apt-get install libheif-examples -y`

Then, I was able to extract the "depth channel" of my iPhone image using the `heif-convert` command and passing the `--with-aux` flag - 
`heif-convert --with-aux -q 90 IMG_1234.heic IMG_1234_out.jpg`

(Replace `IMG_1234` with your actual image name. `-q 90` is optional, and the default quality value. You can change it, the default value is fine though.)

Running it saves multiple images, corresponding to various auxiliary images embedded within the HEIF file. 
I know this is an old post, but I thought I'd add this in case helps anyone else.

**Reply by fmw42** [reply]
See the defines page at https://imagemagick.org/script/defines.php showing:

`heic:depth-image=true	extract the depth image if the container has one.`

**Comment by urban-warrior** [comment, marked as answer]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://www.imagemagick.org/download/beta/ by sometime tomorrow.

**Reply by Milk-Dream-sRGB** [reply]
Wow! Thanks for answer!
But I don't know what is problem, I thinked what I do something wrong or heic have a little support on windows :) Thank you!
If I can ask- what is problem- convert heic to jpg without depth channel or what? Could it be possible to simply extract only depth channel?

PS
I'am sorry- beta ImageMagick 7 or 6? Thanks!

**Reply by Milk-Dream-sRGB** [reply]
And, very sorry for the request, if possible, can I ask for a compiled beta version? (when it's if possible and convenient for you!)
Thank you!

**Comment by Milk-Dream-sRGB** [comment]
I tried compile beta version myself, but it too difficult to me :(
Sorry, if someone can help me I will be happy! :)

**Comment by dlemstra** [comment]
We don't have Windows beta builds available.

**Reply by Milk-Dream-sRGB** [reply]
I had hope... Now it is gone :)
Theoretically, when will the release be?

Сan you advise me why the command "heic-convert" does not work on Windows?

**Comment by Milk-Dream-sRGB** [comment]
I love you. Thanks. <3
